concrete/src/Updater/Migrations/Migrations/Version20190322093418.php

```diff
@@ -0,0 +1,197 @@
+     */
+    private function checkObjectForAttributeClearing(
+        $class,
+        $attributeKeyHandles
+    ) {
+        $em = $this->connection->createEntityManager();
+
+        // For the objects that do not implement the attribute methods on
+        // their entity classes or do not have entity defined for them at
+        // all, we need to do the update manually knowing their context.
+        if ($class === Page::class) {
+            $qb = $this->connection->createQueryBuilder();
+
+            $stmt = $qb->select('p.cID')->from('Pages', 'p')->execute();
+            while ($row = $stmt->fetch()) {
+                $page = Page::getByID($row['cID']);
+
+                $this->clearAttributeValues($page, $attributeKeyHandles);
+            }
+        } elseif ($class === UserInfo::class) {
+            $qb = $this->connection->createQueryBuilder();
+
+            $stmt = $qb->select('u.uID')->from('Users', 'u')->execute();
+            while ($row = $stmt->fetch()) {
+                $ui = $this->app->make(UserInfoRepository::class)
+                    ->getByID($row['uID'])
+                ;
+
+                $this->clearAttributeValues($ui, $attributeKeyHandles);
+            }
+        } else {
+            // For doctrine entities we can do the update more easily
+            // without knowing the context details of each object.
+            $qb = $em->createQueryBuilder();
+            $query = $qb->select('obj')->from($class, 'obj')->getQuery();
+
+            $ir = $query->iterate();
+            foreach ($ir as $subRow) {
+                $this->clearAttributeValues($subRow[0], $attributeKeyHandles);
+            }
+
+            // Execute all pending updates and detach the objects from
+            // Doctrine to preserve memory.
+            $em->flush();
+            $em->clear();
+        }
+    }
+
+    /**
+     * Clears the attribute values for the given object and each of the
+     * attribute keys passed in the second parameter.
+     *
+     * @param AttributeObjectInterface $object the target object to clear the
+     *                                         attributes for
+     * @param string[] $attributeKeyHandles An array of attribute key handles
+     *                                      that need to be cleared
+     */
+    private function clearAttributeValues(
+        AttributeObjectInterface $object,
+        $attributeKeyHandles
+    ) {
+        foreach ($attributeKeyHandles as $akHandle) {
+            // Only clear the attribute if its current value is null. This does
+            // not currently show up in the database search index table making
+            // it impossible to only fetch such objects directly for the update.
+            if ($object->getAttribute($akHandle) === null) {
+                // Set some value for the attribute for it to be
+                // clearable by the indexer.
+                $object->setAttribute($akHandle, 0);
```

Comment:
Scrutinizer seems to give an issue about this call but I think the type hint at `AttributeObjectInterface` is incorrect because the actual implementation allows a `mixed` type for the value:
https://github.com/concrete5/concrete5/blob/1180d447bcd2aa5e8ab9eedbcf28e339f8d211a6/concrete/src/Attribute/ObjectTrait.php#L65